src/rshim.c

```diff
@@ -706,9 +706,18 @@ int rshim_reset_control(rshim_backend_t *bd)
 int rshim_boot_open(rshim_backend_t *bd)
 {
   int rc;
+  time_t t;
 
   pthread_mutex_lock(&bd->mutex);
 
+  /* Add a small delay. */
+  time(&t);
+  if (!bd->register_time || difftime(t, bd->register_time) < 2) {
+    RSHIM_INFO("rshim%d: boot not ready\n", bd->index);
+    pthread_mutex_unlock(&bd->mutex);
+    return -EBUSY;
```

Comment:
-EAGAIN?

Reply:
Not -EAGAIN, because -EAGAIN might cause the caller to retry which is not the intent.

Reply:
I thought the intent is indeed to inform the user to try again later since it'll timeout in 2 seconds, unless it's not a interactive session but a program doing a while-loop trying without delay.

Reply:
There is a mysterious open/close of /dev/rshim0/boot during boot, which caused ARM reset. The goal of this WA is to ignore this request, though more investigation should be needed to find out how this boot file is opened.